From 3b7d1cf250abe96c25fc1941504988df50c17233 Mon Sep 17 00:00:00 2001
From: David Brownlee <abs@absd.org>
Date: Wed, 26 Jul 2023 14:52:44 +0100
Subject: [PATCH] GLStateEGL,GLStateGLX: Change failure to get a "good" visual
 config to warning

With ea488e9 glmark2 was adjusted to abort if it could not find a
"good" EGL or GLX config.

This commit changes that to instead warn and continue in that case.

Fixes glmark2 running on at least NetBSD-10/amd64 on a ThinkPad
T530, and should also address issue#202 for Ubuntu 22.04 on arm64

Fixes #202

Co-authored-by: Alexandros Frantzis <alexandros.frantzis@collabora.com>
---
 src/gl-state-egl.cpp | 8 +++++++-
 src/gl-state-glx.cpp | 8 +++++++-
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/src/gl-state-egl.cpp b/src/gl-state-egl.cpp
index 0d738599..f4239812 100644
--- a/src/gl-state-egl.cpp
+++ b/src/gl-state-egl.cpp
@@ -635,7 +635,13 @@ GLStateEGL::select_best_config(std::vector<EGLConfig>& configs)
         }
     }
 
-    return best_score > 0 ? best_config : 0;
+    if (best_score <= 0) {
+        Log::warning("Unable to find a good EGL config, will continue with the best match,\n"
+                     "but you should verify that the config values are acceptable.\n"
+                     "Tip: Use --visual-config to request a different config\n");
+    }
+
+    return best_config;
 }
 
 bool
diff --git a/src/gl-state-glx.cpp b/src/gl-state-glx.cpp
index 9f41a32b..de1c9f45 100644
--- a/src/gl-state-glx.cpp
+++ b/src/gl-state-glx.cpp
@@ -342,7 +342,13 @@ GLStateGLX::select_best_config(std::vector<GLXFBConfig> configs)
         }
     }
 
-    return best_score > 0 ? best_config : 0;
+    if (best_score <= 0) {
+        Log::warning("Unable to find a good GLX FB config, will continue with the best match,\n"
+                     "but you should verify that the config values are acceptable.\n"
+                     "Tip: Use --visual-config to request a different config\n");
+    }
+
+    return best_config;
 }
 
 GLADapiproc
